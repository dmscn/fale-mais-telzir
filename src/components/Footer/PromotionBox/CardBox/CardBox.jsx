import React, { Component } from 'react'
import classNames from 'classnames';
import Card from "@material-ui/core/Card";

import styles from './CardBox.module.css';

export type Props = {
  className?: string,
  title: string,
  content: string,
  price: number,
};

export default class CardBox extends Component<Props> {
  render() {
    const { className, title, content, price } = this.props;

    return (
      <Card className={classNames(styles.Container, className)}>
        <h3>{title}</h3>
        <p>{content}</p>
        R$ <span className={styles.Price}>{price}</span>
      </Card>
    )
  }
}
