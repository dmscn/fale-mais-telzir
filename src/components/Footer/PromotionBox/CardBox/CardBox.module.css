.Container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  align-items: center;
  width: 15vw;
  height: 15vw;
  margin: .5em;
  transition: all .2s ease-in-out;
}

.Price {
  font-weight: 400;
  color: red;
}