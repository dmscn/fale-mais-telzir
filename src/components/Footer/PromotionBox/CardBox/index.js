export {default} from './CardBox';
export * from './CardBox';