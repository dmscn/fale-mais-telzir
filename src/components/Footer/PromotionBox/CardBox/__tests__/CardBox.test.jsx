import React from "react";
import CardBox from "../CardBox";

describe("<CardBox />", () => {
  it("should render", () => {
    const wrapper = shallow(<CardBox />);
    expect(wrapper).toMatchSnapshot();
  });
});
