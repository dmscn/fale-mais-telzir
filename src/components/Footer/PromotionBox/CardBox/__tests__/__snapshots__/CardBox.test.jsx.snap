// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBox /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h3 />,
        <p>
          Não paga os primeiros 
          <span
            className="MinDescount"
          />
           minutos.
        </p>,
        <p>
          R$ 
          <span
            className="Price"
          />
        </p>,
      ],
      "className": "Container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Não paga os primeiros ",
            <span
              className="MinDescount"
            />,
            " minutos.",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Não paga os primeiros ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "MinDescount",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          " minutos.",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "R$ ",
            <span
              className="Price"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "R$ ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "Price",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h3 />,
          <p>
            Não paga os primeiros 
            <span
              className="MinDescount"
            />
             minutos.
          </p>,
          <p>
            R$ 
            <span
              className="Price"
            />
          </p>,
        ],
        "className": "Container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Não paga os primeiros ",
              <span
                className="MinDescount"
              />,
              " minutos.",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Não paga os primeiros ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "MinDescount",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            " minutos.",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "R$ ",
              <span
                className="Price"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "R$ ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "Price",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
